packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ScmTypeFactory.cs

```diff
@@ -98,8 +98,33 @@ private ExtensibleEnumSerializationProvider[] CreateExtensibleEnumSerializations
                 }
             }
 
-            ClientCache[inputClient] = client;
-            return client;
+            var result = client is not null && IsValidClient(inputClient, client) ? client : null;
+            ClientCache[inputClient] = result;
+            return result;
+        }
+
+        private bool IsValidClient(InputClient inputClient, ClientProvider client)
+        {
+            // client is not valid if it has no operations, sub-client methods, or custom code methods
+            if (inputClient.Operations.Count > 0)
+            {
+                return true;
+            }
+
+            foreach (var subclient in client.SubClients)
+            {
+                if (subclient.Methods.Count > 0)
+                {
+                    return true;
+                }
+            }
+
+            if (client.CustomCodeView?.Methods.Count > 0)
```

Comment:
Checking for the Methods isn't enough because the custom code may be using Properties, Fields, or even just doing something in the constructor. We should simply check whether there is custom code or not, and not worry about the individual members.

Reply:
So, it seems like custom code has a different criteria that if there is any custom code exists we should not skip the client?
Because for client and sub-client, we only check the methods.